server/controllers/batches.ts

```diff
@@ -399,55 +692,82 @@ const generateTeams = (roundGen, users, roundNumber, oldNicks) => {
-    return {users: teamUsers};
+    return { users: teamUsers };
   });
   return teams;
-}
-
-const roundRun = async (batch, users, rounds, i, oldNicks, teamSize, io, kickedUsers) => {
+};
+// if exp is round:
+// [3]
+// [4]
+// that round is unmasked with either 1 / 2's chat history
+const roundRun = async (
+  batch,
+  users,
+  rounds,
+  i,
+  oldNicks,
+  teamSize,
+  io,
+  kickedUsers
+) => {
   const teamFormat = batch.teamFormat;
   // for singleTeamed batches
-  const reconveneBestCondition = batch.randomizeExpRound ? teamFormat === 'single' && batch.expRounds[0] === i + 1 :
-      batch.numRounds === i + 1 && teamFormat === 'single';
-  const reconveneWorstCondition = teamFormat === 'single' &&  batch.worstRounds.length ?
-      batch.worstRounds[0] === i + 1 : false;
+  const reconveneBestCondition = batch.randomizeExpRound
+    ? teamFormat === "single" && batch.expRounds[0] === i + 1
+    : batch.numRounds === i + 1 && teamFormat === "single"; // if randomize, then check if it's the exp round, else, check if it's the last round
+  const reconveneWorstCondition =
+    teamFormat === "single" && batch.worstRounds.length
+      ? batch.worstRounds[0] === i + 1
+      : false;
   if (reconveneBestCondition && reconveneWorstCondition) {
-    console.log('reconvene logic error: running worst round');
+    console.log("reconvene logic error: running worst round");
   }
+  console.log(
+    "reconvene best: ",
+    reconveneBestCondition,
+    "reconvene worst: ",
+    reconveneWorstCondition
+  );
+
   let roundType;
-  if ((!reconveneWorstCondition && !reconveneBestCondition) || batch.bestRoundFunction === 'do not reconvene') {
+  if (
+    (!reconveneWorstCondition && !reconveneBestCondition) ||
+    batch.bestRoundFunction === "do not reconvene"
+  ) {
     roundType = STANDARD;
-  } else {
-    if (reconveneWorstCondition) {
-      roundType = WORST;
-    } else {
-      if (reconveneBestCondition) {
-        roundType = BEST;
-      }
-    }
+  } else if (reconveneWorstCondition) {
+    roundType = WORST;
+  } else if (reconveneBestCondition) {
+    roundType = BEST;
   }
-  let previousRoundWasReconveningBest = batch.randomizeExpRound ? teamFormat === 'single' && batch.expRounds[0] === i :
-      batch.numRounds === i && teamFormat === 'single';
+
   let task;
-  if (teamFormat !== 'single' || roundType === STANDARD) { // standard flow
+
+  if (roundType === WORST) {
+    // worst round: we take one-before-last task
+    task = batch.tasks[batch.tasks.length - 2];
+  } else {
     task = batch.tasks[i];
-  } else { // best round: we take last task
-    if (previousRoundWasReconveningBest) { // we dont take the last task, but the one before last
-      task = batch.tasks[batch.tasks.length - 2]
-    }
-    if (roundType === BEST) {
-      task = batch.tasks[batch.tasks.length - 1];
-    }
-    if (roundType === WORST) { // worst round: we take one-before-last task
-      task = batch.tasks[batch.tasks.length - 2];
-    }
   }
```

Comment:
@xehu see main change here: removed take last task if roundType !== WORST, since this was causing duplicate task issues.